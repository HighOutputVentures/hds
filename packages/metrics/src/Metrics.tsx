import { Box, Button, Flex, HStack, Icon, Text, VStack } from '@chakra-ui/react';
import {
  DotsVerticalIcon,
  NegativeArrowIcon,
  NegativeChartIcon,
  PositiveArrowIcon,
  PositiveChartIcon,
} from '@highoutput/hds-icons';
import * as React from 'react';
import { FC } from 'react';

/** @deprecated */
export interface MetricsProps {
  /** Percentage change of Metric */
  percentage: number;
  /** Field to check total change of the metric positive negative or neutral(no change) */
  metricChange: 'positive' | 'negative';
  /** Main Label of the Metric */
  metricLabel?: string;
  icon?: React.ReactNode;
  metricValue?: string;
  /** Text for when you last viewed the metric */
  lastViewedText?: string;
  /** View Details button Text */
  viewDetailsText?: string;
  /** View Callback function you can add your modal or redirection to page here */
  viewDetailsCallback?: () => {};
  showChart?: boolean;
  __metricButtonTestId?: string;
  __metricValueTestId?: string;
  __metricLabelTestId?: string;
  __metricChartTestId?: string;
}

/** @deprecated */
const Metrics: FC<MetricsProps> = ({
  percentage,
  metricChange,
  metricLabel,
  icon,
  metricValue,
  lastViewedText,
  viewDetailsText,
  viewDetailsCallback,
  showChart,
  __metricButtonTestId,
  __metricValueTestId,
  __metricLabelTestId,
  __metricChartTestId,
}) => {
  return (
    <Flex
      flexDir="column"
      minHeight={114}
      maxW="390px"
      bg="white"
      border="1px solid #FCFCFC"
      padding="20px 16px"
      data-testid="hds.metrics-card.container"
      boxShadow="0px 1px 3px rgba(16, 24, 40, 0.1), 0px 1px 2px rgba(16, 24, 40, 0.06)"
      border-radius="8px"
    >
      <HStack mr={8} w="full" align="start" justify="space-between">
        <HStack>
          {icon}
          <Text
            size="label-sm-default"
            data-testid={__metricLabelTestId ?? 'hds.metrics-card.label'}
          >
            {metricLabel}
          </Text>
        </HStack>
        <Icon as={DotsVerticalIcon} color="neutrals.500" />
      </HStack>

      <HStack w="full" spacing="16px" mt="24px">
        <VStack w="full" align="start" spacing="16px">
          <Text
            mt={2}
            size="header-4"
            data-testid={__metricValueTestId ?? 'hds.metrics-card.value'}
          >
            {metricValue}
          </Text>
          <HStack
            height="18px"
            borderRadius="16px"
            padding="2px 10px 2px 0px"
            data-testid="hds.metrics-card.text"
          >
            {metricChange === 'positive' ? (
              <Icon as={PositiveArrowIcon} />
            ) : (
              <Icon as={NegativeArrowIcon} />
            )}
            <Text
              color={
                metricChange === 'positive'
                  ? 'interface.success.900'
                  : 'interface.error.900'
              }
              size="label-xs-default"
            >
              {percentage}%
            </Text>
            <Text color="neutrals.500">{lastViewedText}</Text>
          </HStack>
        </VStack>
        {showChart && metricChange === 'positive' && (
          <Box
            maxW="128px"
            w="full"
            maxH="64px"
            h="full"
            data-testid={__metricChartTestId ?? 'hds.metrics-card.chart'}
          >
            <Icon as={PositiveChartIcon} w="full" h="full" />
          </Box>
        )}{' '}
        {showChart && metricChange === 'negative' && (
          <Box
            maxW="128px"
            w="full"
            maxH="64px"
            h="full"
            data-testid={__metricChartTestId ?? 'hds.metrics-card.chart'}
          >
            <Icon as={NegativeChartIcon} w="full" h="full" />
          </Box>
        )}
      </HStack>

      {viewDetailsText && (
        <Button
          mt="16px"
          fontWeight="semibold"
          _hover={{
            textDecoration: 'none',
            color: 'brand.primary.700',
          }}
          data-testid={__metricButtonTestId ?? 'hds.metrics-card.button'}
          alignSelf="end"
          onClick={() => {
            if (viewDetailsCallback) viewDetailsCallback();
          }}
          variant="link"
          color="brand.primary.900"
        >
          {viewDetailsText}
        </Button>
      )}
    </Flex>
  );
};

export default Metrics;
