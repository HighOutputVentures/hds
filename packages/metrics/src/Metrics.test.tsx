import { render, fireEvent } from '@testing-library/react';
import Metrics from './Metrics';
import * as React from 'react';

describe('Metrics', () => {
  it('renders correctly with positive metric change', () => {
    const { getByText, container } = render(
      <Metrics
        percentage={5}
        metricChange="positive"
        metricLabel="Revenue"
        metricValue="$1,000"
        lastViewedText="Last viewed: 1 hour ago"
        viewDetailsText="View details"
      />
    );
    expect(container).toMatchSnapshot();
    expect(getByText('Revenue')).toBeDefined();
    expect(getByText('$1,000')).toBeDefined();
    expect(getByText('5%')).toBeDefined();
    expect(getByText('Last viewed: 1 hour ago')).toBeDefined();
    expect(getByText('View details')).toBeDefined();
  });

  it('renders correctly with negative metric change', () => {
    const { getByText, container } = render(
      <Metrics
        percentage={-5}
        metricChange="negative"
        metricLabel="Revenue"
        metricValue="$1,000"
        lastViewedText="Last viewed: 1 hour ago"
        viewDetailsText="View details"
      />
    );
    expect(container).toMatchSnapshot();
    expect(getByText('Revenue')).toBeDefined();
    expect(getByText('$1,000')).toBeDefined();
    expect(getByText('-5%')).toBeDefined();
    expect(getByText('Last viewed: 1 hour ago')).toBeDefined();
    expect(getByText('View details')).toBeDefined();
  });

  it('calls view details callback function on click', () => {
    const viewDetailsCallback = jest.fn();
    const { getByText } = render(
      <Metrics
        percentage={5}
        metricChange="positive"
        metricLabel="Revenue"
        metricValue="$1,000"
        lastViewedText="Last viewed: 1 hour ago"
        viewDetailsText="View details"
        viewDetailsCalllback={viewDetailsCallback}
      />
    );

    fireEvent.click(getByText('View details'));
    expect(viewDetailsCallback).toHaveBeenCalled();
  });
});
