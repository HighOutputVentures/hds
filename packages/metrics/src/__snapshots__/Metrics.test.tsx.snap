// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics renders correctly with negative metric change 1`] = `
<div>
  <div
    border-radius="8px"
    class="css-vi7wl0"
  >
    <div
      class="chakra-stack css-1a8w4xd"
    >
      <div
        class="chakra-stack css-84zodg"
      >
        <p
          class="chakra-text css-0"
        >
          Revenue
        </p>
      </div>
      <svg
        class="chakra-icon chakra-icon css-1gg9e9r"
        focusable="false"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.0001 10.8333C10.4603 10.8333 10.8334 10.4602 10.8334 9.99998C10.8334 9.53974 10.4603 9.16665 10.0001 9.16665C9.53984 9.16665 9.16675 9.53974 9.16675 9.99998C9.16675 10.4602 9.53984 10.8333 10.0001 10.8333Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.66667"
        />
        <path
          d="M10.0001 4.99998C10.4603 4.99998 10.8334 4.62688 10.8334 4.16665C10.8334 3.70641 10.4603 3.33331 10.0001 3.33331C9.53984 3.33331 9.16675 3.70641 9.16675 4.16665C9.16675 4.62688 9.53984 4.99998 10.0001 4.99998Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.66667"
        />
        <path
          d="M10.0001 16.6666C10.4603 16.6666 10.8334 16.2935 10.8334 15.8333C10.8334 15.3731 10.4603 15 10.0001 15C9.53984 15 9.16675 15.3731 9.16675 15.8333C9.16675 16.2935 9.53984 16.6666 10.0001 16.6666Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.66667"
        />
      </svg>
    </div>
    <div
      class="chakra-stack css-s2mkdb"
    >
      <div
        class="chakra-stack css-i1m1ot"
      >
        <p
          class="chakra-text css-g89h0y"
        >
          $1,000
        </p>
        <div
          class="chakra-stack css-bjqmq7"
        >
          <svg
            class="chakra-icon css-13otjrl"
            fill="none"
            focusable="false"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 4.16669V15.8334M10 15.8334L15.8333 10M10 15.8334L4.16667 10"
              stroke="#DC180C"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <p
            class="chakra-text css-1vqbz9l"
          >
            -5
            %
          </p>
          <p
            class="chakra-text css-m6obuh"
          >
            Last viewed: 1 hour ago
          </p>
        </div>
      </div>
    </div>
    <button
      class="chakra-button css-1i9b51t"
      type="button"
    >
      View details
    </button>
  </div>
</div>
`;

exports[`Metrics renders correctly with positive metric change 1`] = `
<div>
  <div
    border-radius="8px"
    class="css-vi7wl0"
  >
    <div
      class="chakra-stack css-1a8w4xd"
    >
      <div
        class="chakra-stack css-84zodg"
      >
        <p
          class="chakra-text css-0"
        >
          Revenue
        </p>
      </div>
      <svg
        class="chakra-icon chakra-icon css-1gg9e9r"
        focusable="false"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.0001 10.8333C10.4603 10.8333 10.8334 10.4602 10.8334 9.99998C10.8334 9.53974 10.4603 9.16665 10.0001 9.16665C9.53984 9.16665 9.16675 9.53974 9.16675 9.99998C9.16675 10.4602 9.53984 10.8333 10.0001 10.8333Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.66667"
        />
        <path
          d="M10.0001 4.99998C10.4603 4.99998 10.8334 4.62688 10.8334 4.16665C10.8334 3.70641 10.4603 3.33331 10.0001 3.33331C9.53984 3.33331 9.16675 3.70641 9.16675 4.16665C9.16675 4.62688 9.53984 4.99998 10.0001 4.99998Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.66667"
        />
        <path
          d="M10.0001 16.6666C10.4603 16.6666 10.8334 16.2935 10.8334 15.8333C10.8334 15.3731 10.4603 15 10.0001 15C9.53984 15 9.16675 15.3731 9.16675 15.8333C9.16675 16.2935 9.53984 16.6666 10.0001 16.6666Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.66667"
        />
      </svg>
    </div>
    <div
      class="chakra-stack css-s2mkdb"
    >
      <div
        class="chakra-stack css-i1m1ot"
      >
        <p
          class="chakra-text css-g89h0y"
        >
          $1,000
        </p>
        <div
          class="chakra-stack css-bjqmq7"
        >
          <svg
            class="chakra-icon css-13otjrl"
            fill="none"
            focusable="false"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 15.8334V4.16669M10 4.16669L4.16666 10M10 4.16669L15.8333 10"
              stroke="#00C408"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <p
            class="chakra-text css-1c6sp0k"
          >
            5
            %
          </p>
          <p
            class="chakra-text css-m6obuh"
          >
            Last viewed: 1 hour ago
          </p>
        </div>
      </div>
    </div>
    <button
      class="chakra-button css-1i9b51t"
      type="button"
    >
      View details
    </button>
  </div>
</div>
`;
